### DISCUSSION TITLE
digitalocean mariadb connection to redash ec2 using ssh

### DISCUSSION THREAD

**Discussion by saurabhhangotra** [getredash/redash#6536] [General]
I have crated a digitalocean mariadb and connecting using ssh
My redash is installed in EC2 so i need to create a connection to connect to mariadb. I tried running below command

ssh -i /path/to/your_ssh_key_file -L 3306:localhost:3306 user_name@ip_address -N

and created new connection with below details

Host: localhost.
Port: 3306
Database Name: Specify the name of your MariaDB database.
Username: Enter your MariaDB username (analytics).
Password: Enter your MariaDB password.

Still not able to connect, can anyone help me on this

**Comment by guidopetri** [comment]
Hey! You might have to look at your firewall settings and make sure both Redash and MariaDB are in the same VPC or are publicly accessible. However, this seems like a digital ocean connectivity problem, so I'd suggest posting there instead.

**Reply by saurabhhangotra** [reply]
I already checked with digital ocean and currently it is exposed to all ipv4 ips and I am able to connect through workbench and with other connectors as well. This seems to be issue with redash only.

**Reply by guidopetri** [reply]
Can you connect from the ec2 instance outside of Redash?

**Reply by saurabhhangotra** [reply]
I didn't checked through EC2 instance but i am able to connect to mariadb through ssh from local powershell as well and through other connectors using ssh

**Reply by guidopetri** [reply]
Can you double check?

**Reply by saurabhhangotra** [reply]
Ya i have checked its working from EC2 but not working in redash connection, any document for ssh connection which i can refer?

**Reply by justinclift** [reply]
To try and figure out what is going wrong, you'll probably need to check the docker logs for the piece that's trying to make the container and failing.

How familiar are you with docker? :smile:
